### ISSUE TITLE
Use python when old ruby installed

### ISSUE THREAD

**Issue by kboone** [junegunn/vim-plug#482]
I am working on a system where my vim is linked against both python and ruby, but the ruby version that is installed is too old for vim-plug (ruby 1.8.5, requirement is 1.8.7). By default, vim-plug tries to use this and crashes. My python version is 2.7, so vim-plug works fine with that if I manually disable the attempt to use ruby in the code.

Could vim-plug be modified to either check for ruby 1.8.7, or have a toggle to force python over ruby?

------------------------------

```
:version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled May 11 2016 10:18:31)
Included patches: 1-1829
Compiled by kboone
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl             +conceal         +file_in_path    +linebreak       -mouse_sysmouse  -python3         -tcl             +wildignore
+arabic          +cryptv          +find_in_path    +lispindent      +mouse_urxvt     +quickfix        +termguicolors   +wildmenu
+autocmd         +cscope          +float           +listcmds        +mouse_xterm     +reltime         +terminfo        +windows
+balloon_eval    +cursorbind      +folding         +localmap        +multi_byte      +rightleft       +termresponse    +writebackup
+browse          +cursorshape     -footer          -lua             +multi_lang      +ruby            +textobjects     +X11
++builtin_terms  +dialog_con_gui  +fork()          +menu            -mzscheme        +scrollbind      +timers          -xfontset
+byte_offset     +diff            +gettext         +mksession       +netbeans_intg   +signs           +title           +xim
+channel         +digraphs        -hangul_input    +modify_fname    +packages        +smartindent     +toolbar         +xsmp_interact
+cindent         +dnd             +iconv           +mouse           +path_extra      +startuptime     +user_commands   +xterm_clipboard
+clientserver    -ebcdic          +insert_expand   +mouseshape      +perl            +statusline      +vertsplit       -xterm_save
+clipboard       +emacs_tags      +job             +mouse_dec       +persistent_undo -sun_workshop    +virtualedit     +xpm
+cmdline_compl   +eval            +jumplist        +mouse_gpm       +postscript      +syntax          +visual
+cmdline_hist    +ex_extra        +keymap          -mouse_jsbterm   +printer         +tag_binary      +visualextra
+cmdline_info    +extra_search    +langmap         +mouse_netterm   +profile         +tag_old_static  +viminfo
+comments        +farsi           +libcall         +mouse_sgr       +python          -tag_any_white   +vreplace
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/home/users/kboone/packages/vim/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/atk-1.0 -I/
usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/libpn
g12     -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc -L/lib64    -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE  -L/home/users/
kboone/packages/anaconda/envs/vim/lib/python2.7/config/ -L/usr/local/lib -Wl,--as-needed -o vim   -L/lib64 -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1
.0 -lgdk_pixbuf-2.0 -lm -lpangocairo-1.0 -lpango-1.0 -lcairo -lgobject-2.0 -lgmodule-2.0 -ldl -lglib-2.0   -lSM -lICE -lXpm -lXt -lX11 -lXdmcp
 -lSM -lICE  -lm -lncurses -lelf -lnsl  -lselinux  -lacl -lattr -lgpm   -Wl,-E -Wl,-rpath,/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/COR
E  /usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/auto/DynaLoader/DynaLoader.a -L/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE -lper
l -lresolv -lnsl -ldl -lm -lcrypt -lutil -lpthread -lc -L/home/users/kboone/packages/anaconda/envs/vim/lib/python2.7/config -lpython2.7 -lpthr
ead -ldl -lutil -lm -Xlinker -export-dynamic   -lruby -lpthread -ldl -lcrypt -lm  -L/usr/lib64

```

- Type:
    - [x] Bug
    - [x] Enhancement
    - [ ] Feature Request
    - [ ] Question
- OS:
    - [ ] All/Other
    - [x] Linux
    - [ ] OS X
    - [ ] Windows
- Vim:
    - [x] Terminal Vim
    - [ ] GVim
    - [ ] Neovim

**Comment by junegunn** [comment]
Does `:ruby puts RUBY_VERSION` print `1.8.5`?

**Comment by kboone** [comment]
Yes it does.

**Comment by junegunn** [comment]
Okay, then we can use the info to reject Ruby installer.

**Comment by junegunn** [comment]
```diff
diff --git a/plug.vim b/plug.vim
index b3bc1a6..15f4e1f 100644
--- a/plug.vim
+++ b/plug.vim
@@ -846,13 +846,14 @@ function! s:names(...)
 endfunction
 
 function! s:check_ruby()
-  silent! ruby require 'thread'; VIM::command('let g:plug_ruby = 1')
-  if get(g:, 'plug_ruby', 0)
-    unlet g:plug_ruby
-    return 1
-  endif
-  redraw!
-  return s:warn('echom', 'Warning: Ruby interface is broken')
+  silent! ruby require 'thread'; VIM::command("let g:plug_ruby = '#{RUBY_VERSION}'")
+  if !exists('g:plug_ruby')
+    redraw!
+    return s:warn('echom', 'Warning: Ruby interface is broken')
+  endif
+  let ruby_version = split(g:plug_ruby, '\.')
+  unlet g:plug_ruby
+  return s:version_requirement(ruby_version, [1, 8, 7])
 endfunction
 
 function! s:update_impl(pull, force, args) abort
```

Can you apply this patch and test it works as expected?

**Comment by kboone** [comment]
Works great, thanks!

**Comment by junegunn** [comment]
Thanks for the confirmation. Fixed in bf1e0fb.
